### DISCUSSION TITLE
Cameras aren't found in Accessory Mode but can be seen if paired through Bridge Mode

### DISCUSSION THREAD

**Discussion by adricrist87** [koush/scrypted#366] [Q&A]
I want to pair my cameras in Accessory mode as suggested to reduce resource usage on my NAS. However, whenever I switch them to Accessory Mode, restart the HomeKit plugin and try to pair the camera, it times out after searching for a while.

I have them paired via bridge mode currently and they work fine (Except for the higher resource consumption)

Here is what I've done to troubleshoot:

- I made sure to have both Home Bridge and Scrypted listening via the same mDNS advertiser Ciao / No luck in ACC Mode
- Made sure Ciao port 5353 was allowed on my Firewall / No luck in ACC Mode
- Made sure the bridge port was allowed on the firewall./ No luck in ACC Mode
- Turned Off my firewall / Worked in ACC mode, but I do not want to have my Firewall off indefinitely.
- Tried Bonjour and Avahi advertisers / No luck in ACC Mode

Any suggestions?

**Comment by koush** [comment]
Are you using the QR code under the settings for the camera in accessory mode?

**Reply by adricrist87** [reply]
Yes I select acc mode, and reset the plug-in, I go back to the camera on the acc mode tab and try to link the camera with the QR code that’s shown there. It recognizes it’s a camera and tells me the whole it’s nos an official accessory or whatnot I select add anyway and it starts searching but it never moves to select the room where the camera is located that’s where it times out.

**Reply by Sco-DUB** [reply]
I'm having the same issue.  Do you find a fix for this?

**Comment by adricrist87** [comment]
Have not found a fix yet. Sorry

**Comment by koush** [comment]
beta plugin now show the ports used by devices in accessory mode.

**Reply by adricrist87** [reply]
![image](https://user-images.githubusercontent.com/93752467/190719685-a91eb87b-8d77-4e81-b1b1-2dfab0f81da7.png)

Is this where the port would be shown? I updated the HomeKit plugin to the 1.1.40 beta and added a camera in accessory mode, It worked fine until I allowed the port shown and turned the firewall on. Then, back to black as before.

I checked the Console and noticed on the output 2 other ports and added them both but these are dynamic either way and when I turned the firewall on it fails to connect.

Something else I noticed In homekit I changed the RTP Sender to Scrypted from ffmpeg thinking it would have better compatibility with the beta but on the device console(firewall off) when the connection ends it says [HomeKit]: ffmpeg exited

**Comment by lox** [comment]
I'm seeing this problem too 🤔 Running scrypted as a Docker container with host network mode and no firewalls involved. Have gotten one camera to pair, but the others fail to pair.

**Comment by lox** [comment]
I've gotten these to pair by switching off Accessory mode, then restarting the homekit plugin and then re-enabling it. They will then pair.
